packages/pyright-internal/src/analyzer/sourceFile.ts

```diff
@@ -421,7 +421,17 @@ export class SourceFile {
                 this._clientDocument = TextDocument.create(this._filePath, 'python', version, '');
             }
             this._clientDocument = TextDocument.update(this._clientDocument, contents, version);
-            this.markDirty();
+
+            const fileContents = this._clientDocument.getText();
+            const contentsHash = StringUtils.hashString(fileContents);
```

Comment:
you could delay calculating the hash till after you check the file length

Reply:
How so? I need to calculate the hash so I can verify that it hasn't changed. If it has changed, I need to use it to update the value in `this._lastFileContentHash`.

Reply:
i was thinking if the file length changed then you wouldn't need to hash it. but I see now it still needs to be stored.